### PR TITLE
Andrew and Tim

### PR DESCRIPTION
**Pull request by AndrewBaik**
#### Single-line Summary
**Today, _ and _ paired together. It took about _**

#### Reflect and summarize on your process for each `TODO` item :  
  1. We identified the taskings and read the starter code.
  2. Refactored forEach to using .map
  3. We've wrapped article.js and articleview.js using app function to render article data.

#### Checklist (before submitting, fill in each set of square brackets with an 'x')
- [x] We have titled the Pull Request similar to our branch name (ex: 'brian-rick'). 
- [x] This PR includes commits from both myself and my partner; e.g. We followed good pair programming practices by switching driver/navigator roles.
- [x] There is no extraneous, unrelated code included in this PR.
- [x] We have summarized our `TODO:` process above.

### CODE REVIEW

File: andrew-tim/public/admin.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <footer>
+      &copy; 2018 Code Fellows | Static Blog | Happy Ipsum!
+    </footer>
+
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.11/handlebars.min.js"></script>
+    <script src="scripts/article.js"></script>
+    <script src="scripts/articleView.js"></script>
+    <script>
+      Article.fetchAll();
```
Comment (sayanything830) [inline andrew-tim/public/admin.html]: `Article .fetchAll` should be accessed as `app.Article.fetchAll()` since both `article.js` and `articleView.js` are wrapped in IIFEs.

`Article.fetchAll` also needs a callback passed in, __*HINT*__ in this case, we want to init the admin page.

File: andrew-tim/public/scripts/article.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        Article.loadAll(results);
+        callback();
+      })
+  };
+  
+  Article.numWordsAll = () => {
+    return Article.all.map(article => article.body.split(' ').length).reduce( function(a, b) {return a + b});
+  };
+  
+  // Article.allAuthors = () => {
```
Comment (sayanything830) [inline andrew-tim/public/scripts/article.js]: Commenting out these lines still breaks your code since the methods are being called in other places.

File: andrew-tim/public/scripts/article.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  //   return Article.all.map( function(a)).reduce();
```
Comment (sayanything830) [inline andrew-tim/public/scripts/article.js]: refactor a bit:

```js
return Article.all.map(article => article.author).reduce(/* pass in two arguments => {
conditional statement if current author does not exist in accumulator authors array, add it
return your accumulated author array
}, [] <-- need this optional argument to declare that your accumulator is an array */);

File: andrew-tim/public/scripts/article.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  // };
+  
+  // Article.numWordsByAuthor = () => {
+  //   return Article.allAuthors().map(author => {})
```
Comment (sayanything830) [inline andrew-tim/public/scripts/article.js]: This one is tricky:

```js
return Article.allAuthors().map(author => {
// returning an object literal
  return {
// with a name property with the author value
    name: author;
// and a numWords property with the total words written by this author
    numwords: Article.all /* filter through all the articles for those only written by the current author, map through each article to get the word count, reduce each word count to return the total word count */
  }
}
```
Note: the property names given here are the ones needed to use as your handlebars template variables.

File: andrew-tim/public/admin.html [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Lab 10</title>
+    <link rel="stylesheet" href="vendor/styles/normalize.css">
+    <link rel="stylesheet" href="vendor/styles/icons.css">
+    <link rel="stylesheet" href="styles/base.css">
+    <link rel="stylesheet" href="styles/layout.css">
+    <link rel="stylesheet" href="styles/modules.css">
+    <link rel="icon" type="image/x-icon" href="https://avatars2.githubusercontent.com/u/18607604?v=3&s=200">
+  </head>
```
Comment (sayanything830) [inline andrew-tim/public/admin.html]: Need your handlebars template script to render author stats

File: andrew-tim/public/scripts/articleView.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+    articleView.handleAuthorFilter();
+    articleView.handleMainNav();
+    articleView.setTeasers();
+    $('pre code').each((i, block) => hljs.highlightBlock(block));
+  };
+
+  articleView.initAdminPage = () => {
+
+    // REVIEW: We use .forEach() here because we are relying on the side-effects of the callback function: appending to the DOM. The callback is not required to return anything.
+
```
Comment (sayanything830) [inline andrew-tim/public/scripts/articleView.js]: need to declare your `template` variable and assign it a value __*HINT*__: see `Article.prototype.toHtml`
